docs/reference/api/metadata-stores/delta.md

```diff
@@ -0,0 +1,95 @@
+---
+title: Delta Lake Metadata Store
+---
+
+# DeltaMetadataStore
+
+::: metaxy.metadata_store.delta.DeltaMetadataStore
+
+## Installation
+
+The Delta Lake backend relies on [`deltalake`](https://delta-io.github.io/delta-rs/python/), which is provided through Metaxy’s `delta` dependency group.
+Install it with:
+
+```bash
+uv sync --group delta
+```
+
+This will pull in the delta-rs bindings alongside the core Metaxy dependencies (if they are not already installed).
+
+## Usage
+
+```py
+from pathlib import Path
+
+import polars as pl
+
+from metaxy.metadata_store.delta import DeltaMetadataStore
+
+root = Path("/data/metaxy/metadata")
+
+with DeltaMetadataStore(
+    root,
+    storage_options={
+        # Optional: forward credentials or configuration to the underlying storage backend
+        "AWS_REGION": "us-east-1",
+    },
+) as store:
+    # Allow writing metadata from integration tests or scripts
+    with store.allow_cross_project_writes():
+        store.write_metadata(
+            MyFeature,
+            pl.DataFrame(
+                {
+                    "sample_uid": [1, 2],
+                    "provenance_by_field": [
+                        {"frames": "hash1", "audio": "hash1"},
+                        {"frames": "hash2", "audio": "hash2"},
+                    ],
+                }
+            ),
+        )
+
+    # Later in the same session (or a new one) you can read lazily
+    lazy_metadata = store.read_metadata(MyFeature)
+    if lazy_metadata is not None:
+        df = lazy_metadata.collect().to_polars()
+        print(df)
+```
+
+### Object Stores
+
+`DeltaMetadataStore` can operate directly against cloud object stores by pointing `root` at a URI such as `s3://bucket/path`, `gs://bucket/path`, or `az://account/container/path`.
+
+```py
+remote_root = "s3://my-metadata-bucket/prod"
+
+store = DeltaMetadataStore(
+    remote_root,
+    storage_options={
+        "AWS_ACCESS_KEY_ID": "...",
+        "AWS_SECRET_ACCESS_KEY": "...",
+        "AWS_REGION": "us-west-2",
+    },
+    object_store_kwargs={
+        "config": {
+            "access_key_id": "...",
+            "secret_access_key": "...",
+            "region": "us-west-2",
+        }
+    },
+)
+```
+
+`storage_options` go straight to `deltalake`.
+
+### Storage Layout
```

Comment:
This is something I would like to have more control over: let's introduce a `layout` option with two possible values: `flat` and `nested`. 

`flat` would be the current implementation. 
`nested` would store deltalake tables `in/feature/key/directories`.

Should only affect how `_feature_uri` is implemented.

Reply:
done